test/geometry/transform/test_thin_plate_spline.py

```diff
@@ -0,0 +1,226 @@
+import pytest
+import kornia as kornia
+import kornia.testing as utils
+
+import torch
+from torch.autograd import gradcheck
+from torch.testing import assert_allclose
+
+
+class TestTransformParameters:
+    @pytest.mark.parametrize('batch_size', [1, 3])
+    def test_smoke(self, batch_size, device, dtype):
+        src = torch.rand(batch_size, 4, 2, device=device)
+        out = kornia.get_tps_transform(src, src)
+        assert len(out) == 2
+
+    @pytest.mark.parametrize('batch_size', [1, 3])
+    def test_no_warp(self, batch_size, device, dtype):
+        src = torch.rand(batch_size, 5, 2, device=device)
+        kernel, affine = kornia.get_tps_transform(src, src)
+        target_kernel = torch.zeros(batch_size, 5, 2, device=device)
+        target_affine = torch.zeros(batch_size, 3, 2, device=device)
+        target_affine[:, [1, 2], [0, 1]] = 1.
+        assert_allclose(kernel, target_kernel, atol=1e-4, rtol=1e-4)
+        assert_allclose(affine, target_affine, atol=1e-4, rtol=1e-4)
+
+    @pytest.mark.parametrize('batch_size', [1, 3])
+    def test_affine_only(self, batch_size, device, dtype):
+        src = torch.tensor([[
+            [0., 0.],
+            [0., 1.],
+            [1., 0.],
+            [1., 1.],
+            [.5, .5],
+        ]], device=device).repeat(batch_size, 1, 1)
+        dst = src.clone() * 2.
+        kernel, affine = kornia.get_tps_transform(src, dst)
+        assert_allclose(kernel, torch.zeros_like(kernel), atol=1e-4, rtol=1e-4)
+
+    @pytest.mark.parametrize('batch_size', [1, 3])
+    def test_exception(self, batch_size, device, dtype):
+        with pytest.raises(TypeError):
+            src = torch.rand(batch_size, 5, 2).numpy()
+            assert kornia.get_tps_transform(src, src)
+
+        with pytest.raises(ValueError):
+            src = torch.rand(batch_size, 5)
+            assert kornia.get_tps_transform(src, src)
+
+    @pytest.mark.grad
+    @pytest.mark.parametrize('batch_size', [1, 3])
+    @pytest.mark.parametrize('requires_grad', [True, False])
+    def test_gradcheck(self, batch_size, device, dtype, requires_grad):
+        src = torch.rand(batch_size, 5, 2, device=device, requires_grad=requires_grad)
+        dst = torch.rand(batch_size, 5, 2, device=device, requires_grad=(not requires_grad))
+        assert gradcheck(kornia.get_tps_transform, (src, dst), raise_exception=True)
```

Comment:
seems that gradient check doesn't work properly. What I would suggest is that you perform the same tests for the intermediate functions in order to find what blocks from backpropagating and run the tests locally.

Reply:
There were a few things I didn't realize about gradcheck, particularly that all the arguments needed to be float64.